torch/nn/modules/module.py

```diff
@@ -729,14 +729,47 @@ def share_memory(self):
     def _get_name(self):
         return self.__class__.__name__
 
-    def __repr__(self):
-        tmpstr = self._get_name() + '(\n'
+    def get_extra_repr(self):
+        r"""Get the extra representation of the module
+
+        To print customized extra information, you should reimplement
+        this method in your own modules. Each item in the returned list will
+        be printed in separate line.
+        """
+        return []
+
+    def _get_extra_lines(self):
+        r"""Helper method to process extra representation"""
+        extra_repr = self.get_extra_repr()
+        if isinstance(extra_repr, str):
+            return extra_repr.split('\n')
+        else:
+            return extra_repr
```

Comment:
I don't think this is necessary. Let's just require it to be a string and split on newlines